src/routers/router.ts

```diff
@@ -84,6 +84,10 @@ export type SwapRoute = {
    * Flag that is true if and only if simulation is requested and simulation fails
    */
   simulationError?: boolean;
+  /**
+   * Flag that is true if and only if simulation is requested and also attempted
+   */
+  simulationAttempted?: boolean;
```

Comment:
What do we need this flag for? Can't the user not having a balance just be part of simulationError? We should try and avoid adding optional flags like this to the returned object- it complicates logic for clients

Reply:
removed it

Reply:
[i made it so that if the swapParams are not set, then it will also set simulationError bit to true](https://github.com/Uniswap/smart-order-router/pull/183/files#diff-2fe4a092f36b8091262f7035638dc00530816382f3a1cca86927a6fd712e9c04R1137)

This will make it easy for web integration. Otherwise, web could request a quote without fromAddress/swapParams set, and they would receive a quote response without simulationError bit set. Based on the logic of only having simulationError and not simulationAttempted, they would think we successfully simulated the gas estimate for it. But that wouldn't be the case.